src/sqlfluff/dialects/dialect_exasol.py

```diff
@@ -188,15 +208,11 @@
         Delimited(Ref("ColumnDatatypeSegment")),
         Ref("UDFParameterDotSyntaxSegment"),
     ),
-    FunctionScriptTerminatorSegment=NamedParser(
+    FunctionScriptTerminatorSegment=TypedParser(
         "function_script_terminator",
         SymbolSegment,
-        name="function_script_terminator",
-        type="function_script_terminator",
-    ),
-    WalrusOperatorSegment=NamedParser(
-        "walrus_operator", SymbolSegment, type="assignment_operator"
     ),
+    WalrusOperatorSegment=StringParser(":=", SymbolSegment, type="assignment_operator"),
```

Comment:
Not related to your changes, but I wonder if we should name this `AssignmentOperatorSegment` for consistency with `type`.

Reply:
I see your point - but if I look at other dialects, they have other `assignment_operator` segments beyond just the walrus, so I think the current name is appropriate. TSQL seems to have _loads_.

Reply:
No worries, I'm not, as they say, _passionate_ about this!